Test matrix for Python 3.6 and 3.7

sciyoshi opened the pull request:
Currently flake8 fails on Python <3.6 due to lack of format strings and variable type annotation syntax, and tests fails on Python 3.6 due to (I think) lack of guaranteed dictionary iteration order.

TkTech commented:
Lets not bother with < 3.6?

> lack of guaranteed dictionary iteration order.

This so recently became a guarantee (instead of a CPython side effect) IMO that we shouldn't depend on it. Practically nothing else that implements the dict interface will have the same guarantee (yet).

sciyoshi commented:
Actually, after digging a bit deeper, the failure on 3.6 is not due to dictionary order, but rather because [`re.split()` doesn't support splitting on empty pattern matches](https://docs.python.org/3/library/re.html#re.split):

> Changed in version 3.7: Added support of splitting on a pattern that could match an empty string.

ericls commented:
@sciyoshi great!
